test/phoenix/router/scope_test.exs

```diff
@@ -205,4 +205,29 @@ defmodule Phoenix.Router.ScopedRoutingTest do
     assert conn.status == 200
     assert conn.resp_body == "api v1 users show"
   end
+
+  test "raises for reserved prefixes" do
+    assert_raise ArgumentError, ~r/`static` is a reserved prefix/, fn ->
+      defmodule StaticController do
```

Comment:
We don't need to create a real controller here as it's not required a compile time